src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -280,19 +283,66 @@ private static boolean isDate(String token) {
   }
 
   /**
-   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
-   * it exists, is added to the builder.
+   * Searches the raw text in the builder for the total price using multiple heuristics. If any find
+   * a price, it is added to the builder.
    */
   private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
-    // Assume that the largest price on the receipt is the total price.
+    if (!findPriceAfterTotal(analysisBuilder)) {
+      findLargestPrice(analysisBuilder);
+    }
+  }
```

Comment:
I guess I could see this getting very nested 

```
if (!a) {
  if (!b) {
    if (!c) {
      d();
    }
  }
}
```

So how about structuring it as 
```
if (a()) {
  return;
}

if (b) {
  return;
} 
```